src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -0,0 +1,92 @@
+import ITechRecordWrapper from "../../../@Types/ITechRecordWrapper";
+import HTTPError from "../../models/HTTPError";
+import {ERRORS, RECORD_COMPLETENESS_ENUM, VEHICLE_TYPE} from "../../assets/Enums";
+import {
+  carCoreMandatorySchema,
+  hgvCoreMandatorySchema, lgvCoreMandatorySchema, motorcycleCoreMandatorySchema,
+  psvCoreMandatorySchema,
+  psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, trlCoreMandatorySchema,
+  trlCoreMandatoryVehicleAttributes
+} from "./CoreMandatoryValidations";
+import {ObjectSchema} from "@hapi/joi";
+import {
+  hgvNonCoreMandatorySchema,
+  psvNonCoreMandatorySchema,
+  trlNonCoreMandatorySchema
+} from "./NonCoreMandatoryValidations";
+
+const validateRecordCompleteness = (validationSchema: ObjectSchema, techRecordFields: any) => {
+  return validationSchema.validate(techRecordFields, {stripUnknown: true});
+};
+
+const validateVehicleAttributes = (vehicleType: string, vehicleAttributes: any) => {
+  if (vehicleType !== VEHICLE_TYPE.TRL) {
+    return validateRecordCompleteness(psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  } else {
+    return validateRecordCompleteness(trlCoreMandatoryVehicleAttributes, vehicleAttributes);
+  }
+};
+
+const validateCoreAndNonCoreMandatoryTechRecordAttributes = (vehicleType: string, techRecordWrapper: ITechRecordWrapper) => {
+  let coreMandatoryValidationResult;
+  let nonCoreMandatoryValidationResult;
+  let coreMandatorySchema: ObjectSchema | any;
+  let nonCoreMandatorySchema: ObjectSchema | any;
+  let techRecord = techRecordWrapper.techRecord[0];
+  if (vehicleType === VEHICLE_TYPE.HGV) {
+    coreMandatorySchema = hgvCoreMandatorySchema;
+    nonCoreMandatorySchema = hgvNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.PSV) {
+    coreMandatorySchema = psvCoreMandatorySchema;
+    nonCoreMandatorySchema = psvNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.TRL) {
+    techRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm} as any;
+    coreMandatorySchema = trlCoreMandatorySchema;
+    nonCoreMandatorySchema = trlNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.LGV) {
+    coreMandatorySchema = lgvCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.CAR) {
+    coreMandatorySchema = carCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.MOTORCYCLE) {
+    coreMandatorySchema = motorcycleCoreMandatorySchema;
+  }
+  coreMandatoryValidationResult = coreMandatorySchema ? validateRecordCompleteness(coreMandatorySchema, techRecord) : undefined;
+  nonCoreMandatoryValidationResult = nonCoreMandatorySchema ? validateRecordCompleteness(nonCoreMandatorySchema, techRecord) : undefined;
```

Comment:
same here as above

Reply:
nonCoreMandatorySchema can be undefined, as LGVs/CARs/MOTOs don't have non-core mandatory fields, so it makes sense that nonCoreMandatorySchema should be undefined for them.